IdentityCore/src/intune/MSIDIntuneMAMResourcesCache.m

```diff
@@ -96,13 +96,14 @@ - (NSString *)resourceForAuthority:(MSIDAuthority *)authority
     return nil;
 }
 
-- (void)setResourcesJsonDictionary:(NSDictionary *)jsonDictionary
+- (BOOL)setResourcesJsonDictionary:(NSDictionary *)jsonDictionary
                            context:(id<MSIDRequestContext>)context
                              error:(NSError **)error
 {
-    if (![self isValid:jsonDictionary context:context error:error]) return;
+    if (![self isValid:jsonDictionary context:context error:error]) return NO;
```

Comment:
same as before, should we check self.dataSource?

Reply:
You can only initialize it with non null datasource.